src/makeInfinityAdder.js

```diff
@@ -1,10 +1,27 @@
 'use strict';
 
 /***
+ *
  * @return {function}
  */
+
 function makeInfinityAdder() {
-  // write code here
+  let result = 0;
+
+  const adder = (arg) => {
+    if (arg === undefined) {
+      const sum = result;
+      result = 0;
+
+      return sum;
+    }
+
+    result += arg;
+
+    return adder;
+  };
```

Comment:
While the implementation is correct, consider naming the inner function something more descriptive than `adder`. Since it's a closure that accumulates a sum, a name like `accumulator` might convey its purpose more clearly.